### PR TITLE
VRF fluid temperature control model crashes when using the Fan:SystemModel

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
VRF fluid temperature control model (AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl) is not compatible with the new Fan:SystemModel. This corrects issue #6218.

Work around: use one of the other available fan models.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [x] CI status: all green or justified
 - [x] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - C++ checks:
   - [x] Argument types
   - [x] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes: n/a
 - [x] Documentation changes in place
 - [ ] Changed docs build successfully
 - n/a ExpandObjects changes?
 - n/a If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
@mjwitte @Myoldmopar easy correction to GetInput. Should be easy review of a few minutes.

**Comment by rraustad** [comment]
So much for getting this right on the first attempt. I'll fix this tonight.

**Comment by rraustad** [comment]
Could someone send me the raw output of the Delight runs so I can compare develop to branch results myself.

**Comment by mjwitte** [comment]
@rraustad I wouldn't worry about the delight failures.

**Comment by rraustad** [comment]
This one is done and done.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 1]
```diff
@@ -3428,7 +3432,11 @@ namespace HVACVariableRefrigerantFlow {
 								DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).VRFIUPtr = VRFTUNum;
 								DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).VRFOUPtr = VRFTU( VRFTUNum ).VRFSysNum;
 								DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).SupplyFanIndex = VRFTU( VRFTUNum ).FanIndex;
-								DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).RatedAirVolFlowRate( 1 ) = EnergyPlus::Fans::Fan( VRFTU( VRFTUNum ).FanIndex ).MaxAirFlowRate;
+								if ( VRFTU( VRFTUNum ).fanType_Num == DataHVACGlobals::FanType_SystemModelObject ) {
+									DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).RatedAirVolFlowRate( 1 ) = HVACFan::fanObjs[ VRFTU( VRFTUNum ).FanIndex ]->designAirVolFlowRate;
+								} else {
+									DXCoils::DXCoil( VRFTU( VRFTUNum ).HeatCoilIndex ).RatedAirVolFlowRate( 1 ) = EnergyPlus::Fans::Fan( VRFTU( VRFTUNum ).FanIndex ).MaxAirFlowRate;
+								}
```
Comment (rraustad) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:3439]: These are the only 2 places where the fan is called without checking fan type. The rest of this change is documentation and unit test.

File: tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc [hunk 2]
```diff
@@ -1615,13 +1717,27 @@ namespace EnergyPlus {
...
+		DataZoneEquipment::ZoneEquipConfig( 1 ).IsControlled = true;
+		DataZoneEquipment::ZoneEquipConfig( 1 ).NumInletNodes = 1;
+		DataZoneEquipment::ZoneEquipConfig( 1 ).NumExhaustNodes = 1;
+		DataZoneEquipment::ZoneEquipConfig( 1 ).InletNode.allocate( 1 );
+		DataZoneEquipment::ZoneEquipConfig( 1 ).ExhaustNode.allocate( 1 );
+		DataZoneEquipment::ZoneEquipConfig( 1 ).InletNode( 1 ) = 2;
+		DataZoneEquipment::ZoneEquipConfig( 1 ).ExhaustNode( 1 ) = 1;
+
 		GetVRFInputData( ErrorsFound ); // read VRF
-		// EXPECT_FALSE( ErrorsFound );
```
Comment (rraustad) [inline tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:1619]: This was sort of a weak unit test where GetVRFInputData didn't get everything needed for a complete VRF system. It only got the VRF condenser. I added the rest of the system and now ErrorsFound is false and the Fan:SystemModel causes no errors.

**Comment by mjwitte** [comment]
@rraustad Merged in develop and fixed some minor doc problems.  For some reason, AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl didn't want to hyphenate, so it looked like this.
image
I added some optional hyphens and now it looks like this.
image
@Myoldmopar If there's a better solution, let me know.  Otherwise I'm ready to merge this.  Defect file runs now without crashing.

**Comment by Myoldmopar** [comment]
Yes, this is a problem with our very long object names.  LaTeX does not understand that a colon is a valid character to hyphenate on, and no such rule exists.  I'm curious about trying something however.  Could you add this line right above that paragraph and see if it hyphenates?

```
\hyphenation{AirConditioner:-VariableRefrigerantFlow:-FluidTemperatureControl}
```

It _should_ tell LaTeX that it allows hyphenation after each colon.  If that works, then I know what my next fun little task is going to be.  Automatically generating an object list from the current IDD and including this list of `hyphenation` commands into all the docs.  It would make the docs look *much* better....I think.  I know that breaking the object names will be kind of funny, so the typography folks around here may dislike it.  But given that we simply _have_ super long names, it's probably better to go ahead and hyphenate rather than just leave them going off the page.

**Comment by Myoldmopar** [comment]
Nevermind, it won't work.  `\hyphenate` won't accept non-alpha characters.  The `hyphenat` package could do this it looks like, but adding a package and such is well beyond the scope here.  I'd say you could either manually break that long string into individual words (AirConditioner: VariableRefrigerantFlow: FluidTemperatureControl -- the user will understand this) or just leave it alone.

**Comment by mjwitte** [comment]
@Myoldmopar integration.5ZoneAirCooledWithDOASAirLoop is failing only for the [linux integration coverage](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-9d65cdb78b90b36fe39d3059720a0ffbfbda0825-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html).  It's failing in develop as well.  I can't reproduce the failure with Win64 debug.  Can you get more details?

**Pull request merged**